### PR TITLE
Allow name parameter

### PR DESCRIPTION
**Pull request by badgerspoke**
*Issue #, if available:* https://github.com/aws-samples/copy-encrypted-ami/issues/12

*Description of changes:*
- add option to pass name for new AMI
- only tag new AMI if source AMI has tags

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: copy_encrypted_ami.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ command -v aws >/dev/null 2>&1 || die "aws cli is required but not installed. Ab
 
 
 
-while getopts ":s:d:a:l:r:k:nth" opt; do
+while getopts ":s:d:a:N:l:r:k:nth" opt; do
```
Comment (rollucas) [inline copy_encrypted_ami.sh]: Could you also add in the usage description in the usage function ?

File: copy_encrypted_ami.sh [hunk 2]
```diff
@@ -210,7 +210,9 @@ echo -e "${COLOR}AMI created succesfully in the destination account:${NC} ${CREA
 # Copy Tags
 if [ "${TAG_OPT}x" != "x" ]; then
     AMI_TAGS=$(echo ${AMI_DETAILS} | jq '.Tags')"}"
-    NEW_AMI_TAGS="{\"Tags\":"$(echo ${AMI_TAGS} | tr -d ' ')
-    $(aws ec2 create-tags --resources ${CREATED_AMI} --cli-input-json ${NEW_AMI_TAGS} --profile ${DST_PROFILE} --region ${DST_REGION} || die "Unable to add tags to the AMI in the destination account. Aborting.")
-    echo -e "${COLOR}Tags added sucessfully${NC}"
+    if [ "${AMI_TAGS}x" != "x" ]; then
```
Comment (rollucas) [inline copy_encrypted_ami.sh]: If the user has used "-t" but the source has no tags, this block should not execute at all. As the response from line 22 (if you remove the "}" bit ) will be null, I would change that this line here to:
if [ "${AMI_TAGS}" != "null" ]; then

File: copy_encrypted_ami.sh [hunk 3]
```diff
@@ -210,7 +210,9 @@ echo -e "${COLOR}AMI created succesfully in the destination account:${NC} ${CREA
 # Copy Tags
 if [ "${TAG_OPT}x" != "x" ]; then
     AMI_TAGS=$(echo ${AMI_DETAILS} | jq '.Tags')"}"
```
Comment (rollucas) [inline copy_encrypted_ami.sh]: Remove "}" , see comment below

File: copy_encrypted_ami.sh (hunk 1 above)
Comment (badgerspoke) [reply copy_encrypted_ami.sh]: Oh, yes sorry - done

File: copy_encrypted_ami.sh (hunk 2 above)
Comment (badgerspoke) [reply copy_encrypted_ami.sh]: Hmm, you're right we do get a null - but the `}` is there in the original file before my modifications and exists to ensure the later tagging operation has valid JSON (though it's a bit clunky IMHO)

File: copy_encrypted_ami.sh (hunk 3 above)
Comment (badgerspoke) [reply copy_encrypted_ami.sh]: See above :)

**Comment by rollucas** [comment]
Looks good to me

**Pull request merged**